aws-ssm-resourcedatasync/src/main/java/software/amazon/ssm/resourcedatasync/CreateHandler.java

```diff
@@ -149,7 +149,8 @@ private boolean isRDSStabilized(final ResourceModel model, final AmazonWebServic
                 // syncFromSource operation does not have currentStatus attribute
                 return true;
             }
-            if (currentStatus != null && currentStatus.equals(LastResourceDataSyncStatus.SUCCESSFUL)) {
+            if (currentStatus != null && (currentStatus.equals(LastResourceDataSyncStatus.SUCCESSFUL)
+                    || currentStatus.equals(LastResourceDataSyncStatus.INPROGRESS))) {
                 return true;
             }
             if (currentStatus != null && currentStatus.equals(LastResourceDataSyncStatus.FAILED)) {
```

Comment:
nit : Not a blocker but reommend extracting failure condition into a method. Check if it has failed and return true otherwise